packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -105,7 +121,28 @@ export class AtomStepsModal {
     e.stopImmediatePropagation()
     this.isOpen = false
     this.atomDidDismiss.emit(this.currentStep)
-    this.currentStep = 1
+    this.currentStep = this.customInitialStep ?? 1
+  }
+
+  private get secondaryButtonText() {
+    return this.currentStep === this.customInitialStep
+      ? this.cancelButtonText
+      : this.secondaryButtonTextsArray[this.currentStep - 1].trim()
+  }
+
+  private get primaryButtonText() {
+    return this.primaryButtonTextsArray[this.currentStep - 1].trim()
+  }
+
+  private get progress() {
+    if(this.customInitialStep) {
+      const currentStepAdjustedAsInitial = this.currentStep - this.customInitialStep + 1
+      const stepsQuantityAjustedToCustomInitial = (this.steps + 1 - this.customInitialStep)
+
+      return ( currentStepAdjustedAsInitial / stepsQuantityAjustedToCustomInitial )
```

Comment:
lint: didn't work automatically?

we also could use `early return`

```suggestion
    if (!this.customInitialStep)
    
    const currentStepAdjustedAsInitial = this.currentStep - this.customInitialStep + 1
    const stepsQuantityAjustedToCustomInitial = (this.steps + 1 - this.customInitialStep)

    return (currentStepAdjustedAsInitial / stepsQuantityAjustedToCustomInitial)
```

Reply:
To make the calculation more simple, doesn't it make sense to receive the `customInitialStep` without having to pass the number as an index? Since the entire calculation is not based on the index.

Reply:
maybe we don't need a diferrent calculation, Isn't currentStep already set to use customInitialStep?

Reply:
the `customInitialStep` feature was made to ignore the previous steps existing before the index of this prop, so if I create a steps modal with 4 steps, but set `customInitialStep` as 3, the first step showing when opening the modal should be 3 and, as now, the modal, visually only have 2 steps, the progress should change, instead of being a 4 step progress (1/4+2/4+3/4+4/4) it should be only a 2 step progress (1/2+2/2), so to maintain the coherence between this change, this calculation is needed


And about what @felipefialho about early return, i didn't see any diff between this two approaches, since we need a return of the adjusted progress and a return of the default progress, currently I'm already using early return

Reply:
>the customInitialStep feature was made to ignore the previous steps existing before the index of this prop, so if I create a steps modal with 4 steps, but set customInitialStep as 3, the first step showing when opening the modal should be 3 and, as now, the modal, visually only have 2 steps, the progress should change, instead of being a 4 step progress (1/4+2/4+3/4+4/4) it should be only a 2 step progress (1/2+2/2), so to maintain the coherence between this change, this calculation is needed

@jowjow22  But anyway, it’s still a 4-step modal, right? Or is the idea actually to cut the previous steps?

Reply:
@felipefialho did the early return here [a44c81d](https://github.com/juntossomosmais/atomium/pull/627/commits/a44c81d727fc3605a6e972dfcaa8d78cc2fdffaa)

Reply:
@jowjow22 Oh, maybe we could change the name for: `startModalOnStep`. Currently, I have the impression that will be the current step.